src/crypto/tls/handshake_client_tls13.go

```diff
@@ -571,7 +571,9 @@ func (hs *clientHandshakeStateTLS13) establishHandshakeKeys() error {
 			c.sendAlert(alertIllegalParameter)
 			return fmt.Errorf("%s decaps: %w", sk.Scheme().Name(), err)
 		}
-	} else {
+	}
+
+	if sharedKey == nil {
```

Comment:
This only needs to be part of the `ecdheParameters` branch, but it does not hurt to keep it here in order to reduce the delta with upstream.

Reply:
Indeed